hugr/src/types/type_param.rs

```diff
@@ -323,13 +365,21 @@ pub fn check_type_arg(arg: &TypeArg, param: &TypeParam) -> Result<(), TypeArgErr
             _,
         ) if param.contains(cached_decl) => Ok(()),
```

Comment:
Not by current canonicalization as `TypeArg::Type(Type::RowVariable)` (which is to say, `cached_decl` cannot be a `TypeParam::List { TypeParam::Type }` - and I wasn't planning on doing row-variables that aren't types, the use cases are tiny. If we canonicalized in the other direction, hmmm, maybe that would fall out, hmmm)

Reply:
However ATM I am leaning against canonicalizing `TypeArg::Type(Type::RowVariable)` to `TypeArg::Variable(...TypeParam::List(TypeParam::Type))`, or indeed `TypeArg::Type(Type::Variable)` to `TypeArg::Variable(TypeParam::Type)` because (as well as then needing a TypeNotVariable to go inside `TypeArg::Type`) we would *still* need to keep `TypeArgVariable` to rule out `TypeArg::Variable(TypeParam::Extensions)` which should be an `ExtensionSet::new_type_var`